src/Stack2nix/External/Cabal2nix.hs

```diff
@@ -1,24 +1,29 @@
 {-# LANGUAGE RecordWildCards   #-}
+{-# LANGUAGE OverloadedStrings #-}
 
 module Stack2nix.External.Cabal2nix (
   cabal2nix
   ) where
 
-import           Cabal2nix                   (cabal2nixWithDB)
+import           Cabal2nix                   (cabal2nixWithDB, parseArgs, optNixpkgsIdentifier)
+import           Control.Lens
 import           Data.Maybe                  (fromMaybe)
 import           Data.Text                   (Text, unpack)
+import qualified Distribution.Nixpkgs.Haskell.Hackage as DB
+import           Distribution.Nixpkgs.Haskell.Derivation (Derivation)
 import           Distribution.System         (Platform(..), Arch(..), OS(..))
+import           Language.Nix
 import           System.IO                   (hPutStrLn, stderr)
 import           Stack2nix.Types             (Args (..))
-import qualified Distribution.Nixpkgs.Haskell.Hackage as DB
-import           Distribution.Nixpkgs.Haskell.Derivation (Derivation)
+
 import           Text.PrettyPrint.HughesPJClass (Doc)
 
 cabal2nix :: Args -> FilePath -> Maybe Text -> Maybe FilePath -> DB.HackageDB -> IO (Either Doc Derivation)
 cabal2nix Args{..} uri commit subpath hackageDB = do
   let runCmdArgs = args $ fromMaybe "." subpath
   hPutStrLn stderr $ unwords ("+ cabal2nix":runCmdArgs)
-  cabal2nixWithDB hackageDB runCmdArgs
+  options <- parseArgs runCmdArgs
+  cabal2nixWithDB hackageDB (options { optNixpkgsIdentifier = \i -> Just (binding # (i, path # ["pkgs", i])) })
```

Comment:
So instead of using `attr` it will choose `pkgs.attr`?
Could this update of options be put into a separate function just to make it easier to understand?

Reply:
Sure :)